src/he/conn.c

```diff
@@ -456,6 +457,59 @@ he_return_code_t he_conn_send_keepalive(he_conn_t *conn) {
   return he_internal_send_message(conn, (uint8_t *)&ping, sizeof(he_msg_ping_t));
 }
 
+// Returns true if current state is valid for sending/receiving the HE_MSGID_SERVER_CONFIG message
+bool he_internal_is_valid_state_for_server_config(he_conn_t *conn) {
+  if(conn == NULL) {
+    return false;
+  }
+
+  // The server config message is only valid after the TLS link is established
```

Comment:
Currently all of our messages need to be done after we've authenticated. It's probably not an issue but we might want to tighten this up a little bit.